packages/grafana-i18n/src/i18n.tsx

```diff
@@ -11,22 +11,49 @@ import { TransProps, TransType } from './types';
 let tFunc: TFunction<string[], undefined> | undefined;
 let transComponent: TransType;
 
-export async function initPluginTranslations(id: string) {
+(async () => {
+  // We want to translate grafana-ui without introducing any breaking changes for consumers
+  // who use grafana-ui outside of grafana (such as grafana.com self serve). The other struggle
+  // is that grafana-ui does not require a top-level provider component, so we don't get the
+  // chance to do the mandatory i18next setup that <Trans /> and t() requires
+  const isGrafanaContext = Object.hasOwn(window, 'grafanaBootData');
+  const isTestContext = process.env.NODE_ENV === 'test';
+
+  if (isGrafanaContext || isTestContext) {
+    return;
+  }
+
+  console.log('Initializing a default i18n instance');
+  await initPluginTranslations('unkown-context');
+})();
```

Comment:
Using a hook in `useTranslate` and `<Trans>` to call `initPluginTranslations` resulted in lots of unit test errors that depend on no re-renders which would be the side effect of calling an async function in a hook. I'm really open for other solutions.